codechecker_lib/log_parser.py

```diff
@@ -15,15 +18,80 @@
 
 
 # -----------------------------------------------------------------------------
-def parse_compile_commands_json(logfile):
+def get_compiler_includes(compiler):
+    """
+    Returns a list of default includes of the given compiler.
+    """
+    LOG.debug_analyzer('getting include paths for  ' + compiler)
+    start_mark = "#include <...> search starts here:"
+    end_mark = "End of search list."
+
+    cmd = compiler + " -E -x c++ - -v "  # what if not c++?
+    include_paths = []
+    try:
+        proc = subprocess.Popen(shlex.split(cmd),
+                                stdin=subprocess.PIPE,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+
+        out, err = proc.communicate("")
+
+        print_line = False
+        for line in err.splitlines(True):
+            if line.startswith(end_mark):
+                print_line = False
+            if print_line:
+                include_paths.append("-I"+line.strip())
+            if line.startswith(start_mark):
+                print_line = True
+
+    except OSError as oerr:
+        LOG.error("Cannot find include paths:" + oerr.strerror+"\n")
+
+    finally:
+        return include_paths
+
+
+# -----------------------------------------------------------------------------
+def get_compiler_defines(compiler):
+    """
+    Returns a list of default defines of the given compiler.
+    """
+    cmd = compiler + " -dM -E -"
+    defines = []
+    FNULL = open(os.devnull, 'r')
+    try:
+        proc = subprocess.Popen(shlex.split(cmd),
+                                stdin=FNULL,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+        out, err = proc.communicate("")
+        for line in out.splitlines(True):
+            LOG.debug("define:"+line)
+            define = line.strip().split(" ")[1:]
+            d = "-D"+define[0] + '=' + '"' + ' '.join(define[1:]) + '"'
+            defines.append(d)
+
+    except OSError as oerr:
+        LOG.error("Cannot find defines:" + oerr.strerror+"\n")
+    return defines
```

Comment:
`finally` can be used here too like in the `get_compiler_includes` method before.